app/routes.py

```diff
@@ -0,0 +1,357 @@
+
+    if available_inventory == 0:
+        return jsonify(message="Could not perform checkout"), 400
+
+    new_rental_check_out= Rental(
+        customer_id=customer.id,
+        video_id=video.id,
+        due_date=(datetime.now() + timedelta(days=7)),
+        checked_out=True
+    )
+
+    db.session.add(new_rental_check_out)
+    db.session.commit()
+
+    available_inventory -= 1
+
+    videos_customer_checked_out = Rental.query.filter_by(customer_id=customer.id, checked_out=True).count()
+    return {
+        "customer_id": customer.id,
+        "video_id": video.id,
+        "due_date": new_rental_check_out.due_date,
+        "videos_checked_out_count": videos_customer_checked_out,
+        "available_inventory": available_inventory
+    }, 200
+
+
+@rentals_bp.route("/check-in", methods = ["POST"])
+def rental_check_in():
+    rental_request = request.get_json()
+
+    if "customer_id" not in rental_request or "video_id" not in rental_request:
+        return jsonify(details="Request body must include customer id and video id"), 400
+
+    customer_id = rental_request["customer_id"]
+    video_id = rental_request["video_id"]
+
+    try: 
+        video_id = int(video_id)
+        customer_id = int(customer_id)
+    except:
+        return jsonify(None), 400
+
+    customer = Customer.query.get(customer_id)
+    video = Video.query.get(video_id)
+
+    if customer is None or video is None:
+        return jsonify(None), 404
+    
+    rental = Rental.query.filter_by(customer_id=customer.id, video_id=video.id, checked_out=True).first()
+    if rental is None:
+        return jsonify(message=f"No outstanding rentals for customer {customer.id} and video {video.id}"), 400
+
+    rental.checked_out = False
+    db.session.commit()
+
+    num_currently_checked_out = Rental.query.filter_by(video_id=video.id, checked_out=True).count()
+    available_inventory = video.total_inventory - num_currently_checked_out
+    videos_customer_checked_out = Rental.query.filter_by(customer_id=customer.id, checked_out=True).count()
+    
+    return {
+        "customer_id": customer.id,
+        "video_id": video.id,
+        "videos_checked_out_count": videos_customer_checked_out,
+        "available_inventory": available_inventory
+    }, 200
+
+@customers_bp.route("/<customer_id>/rentals", methods=["GET"]) 
+def get_customers_current_rentals(customer_id):
+    """
+    List of videos a customer currently has checked out
+    """
+    try: 
+        customer_id = int(customer_id)
+    except:
+        return jsonify(None), 400
+    
+    customer = Customer.query.get(customer_id)
+    
+    if customer == None:
+        return jsonify(message=f"Customer {customer_id} was not found"), 404
+
+    rental_list = Rental.query.filter_by(customer_id=customer.id, checked_out=True)
+
+    list_of_dicts = []
+
+    for rental in rental_list:
+        
+        video = Video.query.get(rental.video_id)
```

Comment:
Even though we can't use the default relationships to get the current checked out rentals for customers or videos, they would still help us here, by letting us write `rental.video.title` or `rental.video.release_date` directly, without needing to do the lookup ourselves. As mentioned elsewhere, those relationships can interact badly with secondary relationships, which is why I would tend to avoid the secondary relationship in this project and use the regular one-to-may relationships.